docs/content/en/docs/reference/clusterspec/baremetal.md

```diff
@@ -12,13 +12,577 @@ The following additional optional configuration can also be included:
+* environment.CONTENTS.warnings: Sets dsid_missing_source to `off`.
+* environment.DEST_DISK: Destination block storage device partition where the operating system is located (`/dev/sda2`, by default).
+* environment.DEST_PATH: Location of the cloud-init configuration file on disk (`/etc/cloud/cloud.cfg.d/10_tinkerbell.cfg`, by default)
+* environment.DIRMODE: Linux directory permissions bits to use when creating directories (0700, by default)
+* environment.FS_TYPE: Type of filesystem on the partition (ext4, by default).
+* environment.GID: The Linux group ID to set on file. Set to 0 (root group) by default.
+* environment.MODE: The Linux permission bits to set on file (0600, by default).
+* environment.UID: The Linux user ID to set on file. Set to 0 (root user) by default.
+* image: Container image used to perform the steps needed by this action.
+* timeout: Time needed to complete the action, in seconds.
+
+### template.tasks.actions.add-tink-cloud-init-ds-config (Ubuntu)
+The `add-tink-cloud-init-ds-config` action configures cloud-init data store features. This identifies the location of your metadata source once the machine is up and running. It identifies:
+
+* environment.CONTENTS.datasource: Sets the datasource. Uses Ec2, by default.
+* environment.DEST_DISK: Destination block storage device partition where the operating system is located (/dev/sda2, by default).
+* environment.DEST_PATH: Location of the data store identity configuration file on disk (/etc/cloud/ds-identify.cfg, by default) 
+* environment.DIRMODE: Linux directory permissions bits to use when creating directories (0700, by default)
+* environment.FS_TYPE: Type of filesystem on the partition (ext4, by default).
+* environment.GID: The Linux group ID to set on file. Set to 0 (root group) by default.
+* environment.MODE: The Linux permission bits to set on file (0600, by default).
+* environment.UID: The Linux user ID to set on file. Set to 0 (root user) by default.
+* image: Container image used to perform the steps needed by this action.
+* timeout: Time needed to complete the action, in seconds.
+
+### template.tasks.actions.kexec-image (Ubuntu)
+The `kexec-image` action performs provisioning activities on the machine, then allows kexec to pivot the kernel to use the system installed on disk. This action identifies:
+
+* environment.BLOCK_DEVICE: Disk partition on which the operating system is installed (/dev/sda2, by default)
+* environment.FS_TYPE: Type of filesystem on the partition (ext4, by default).
+* image: Container image used to perform the steps needed by this action.
+* pid: Process ID. Set to host, by default.
+* timeout: Time needed to complete the action, in seconds.
+* volumes: Identifies mount points that need to be remounted to point to locations in the installed system.
+
+If your hardware requires a full reboot, you can change the kexec-image setting as follows:
```

Comment:
Do we want to mention that there are known issues with kexec due to idiosyncratic drivers or drivers that perform DMA etc. If they run into that, reboot action is the way to go.